libdnf/comps/environment/query.cpp

```diff
@@ -63,25 +65,29 @@ EnvironmentQuery::EnvironmentQuery(const BaseWeakPtr & base) : base(base) {
             continue;
         }
 
+        repoid = solvable->repo->name;
+
         // Add installed environments directly, because there is only one solvable for each
-        if (strcmp(solvable->repo->name, "@System") == 0) {
+        if (strcmp(repoid, "@System") == 0) {
             Environment environment(base);
             environment.environment_ids.push_back(EnvironmentId(solvable_id));
             add(environment);
         } else {
             // Create map of available environments:
             // for each environmentid (SOLVABLE_NAME), list all corresponding solvable_ids
-            // TODO(pkratoch): Sort solvable_ids for each environmentid according to repo id
             available_map[solvable_name_pair.second].insert(
-                available_map[solvable_name_pair.second].begin(), solvable_id);
+                available_map[solvable_name_pair.second].end(), std::pair<Id, std::string_view>(solvable_id, repoid));
```

Comment:
... and here you could use `std::make_pair()`, which is a bit less verbose way to construct a pair (you don't need to specify the template arguments, they are inferred). If `repoid` was `std::string_view` :slightly_smiling_face:.